### PR TITLE
1. add go module support for sample-controller/hack/update-codegen.sh

### PR DESCRIPTION
**Pull request by bigpigeon**
**What type of PR is this?**

/kind bug

**Which issue(s) this PR fixes**:

Fixes #

project https://github.com/kubernetes/sample-controller file` hack/update-codegen.sh ` cannot run in go module 

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/sample-controller/hack/update-codegen.sh [hunk 1]
```diff
@@ -19,7 +19,13 @@ set -o nounset
 set -o pipefail
 
 SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+
+if [[ $(go env GO111MODULE) == "on" ]]; then
```
Comment (liggitt) [inline staging/src/k8s.io/sample-controller/hack/update-codegen.sh:23]: don't key off of explicit envvar values… this changes let go version. see suggestions at https://github.com/kubernetes/kubernetes/pull/85559#issuecomment-570234199 (which is a substantially similar PR that seems to have gone stale)

Comment (bigpigeon) [reply staging/src/k8s.io/sample-controller/hack/update-codegen.sh:24]: Definition of CODEGEN_PKG moved into if-else condition

**Comment by liggitt** [comment]
closing this in favor of https://github.com/kubernetes/kubernetes/pull/93728

**Comment by makusu2** [comment]
@liggitt Could this be reopened, given that the PR you linked has now been closed without merging?

**Comment by liggitt** [comment]
revived #93728 in https://github.com/kubernetes/kubernetes/pull/103847

**Pull request closed without merging**